### PR TITLE
Use GUI font if spectator string contains special chars.

### PR DESCRIPTION
**Pull request by VelocityRa**
For #507.
Also made some `ScoreboardView` methods `const`.

### CODE REVIEW

File: Sources/Client/ScoreboardView.cpp [hunk 1]
```diff
@@ -49,17 +49,33 @@ namespace spades {
...
+			tc = nullptr;
+			ctf = nullptr;
+			image = nullptr;
+
+			// Use GUI font if spectator string has special chars
+			auto spectatorString = _TrN("Client", "Spectator{1}", "Spectators{1}", "", "");
+			auto has_special_char = 
+				std::find_if(spectatorString.begin(), spectatorString.end(),
+				[](char ch) {
+					ch = static_cast<unsigned char>(ch);
```
Comment (yvt) [inline Sources/Client/ScoreboardView.cpp]: I don't think this cast is necessary...

Comment (VelocityRa) [inline Sources/Client/ScoreboardView.cpp]: It is, the `char` `isalnum` should be unsinged. A runtime error occurs otherwise.

Comment (yvt) [inline Sources/Client/ScoreboardView.cpp]: Okay, I see, `isalnum` accepts a `int` and it must be representable as `unsigned char` (or is equal to `EOF`). But casting `ch` to `unsigned char` and then reassigning the result to `ch` doesn't make `ch` unsigned. In fact, this is what I get from LLDB:

```
(lldb) bt 1
* thread #1: tid = 0x1054e35, 0x00000001003733c2 OpenSpades`isalnum(_c=-24) + 18 at ctype.h:214, queue = 'com.apple.main-thread', stop reason = step in
    frame #0: 0x00000001003733c2 OpenSpades`isalnum(_c=-24) + 18 at ctype.h:214
```

**Comment by yvt** [comment]
Also, can you make the text a little bit smaller to match the original font, and move it downward? The best way to add a new font size would be adding a new member function to `FontManager` (in `Fonts.h`).

File: Sources/Client/ScoreboardView.cpp (hunk 1 above)
Comment (VelocityRa) [inline Sources/Client/ScoreboardView.cpp]: Still, this line does _something_ that makes it work. Without it I get this when I have a font with non-ascii chars:
image
And with it it works as intended.

**Comment by VelocityRa** [comment]
Can I make a new IFont member `mediumFont`(or something similar) instead? I don't have much time for this right now.

**Comment by yvt** [comment]
Okay

**Comment by VelocityRa** [comment]
@yvt How's that?
image

**Comment by yvt** [comment]
@VelocityRa That looks great!

**Comment by VelocityRa** [comment]
Cool, I force pushed to the previous commit (and made a seperate one for the font).

**Pull request merged**